src/products/products.service.ts

```diff
@@ -0,0 +1,91 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { Model } from 'mongoose';
+import { Product } from './Product.model';
+
+@Injectable()
+export class ProductsService {
+  constructor(
+    @InjectModel('Product') private readonly productModel: Model<Product>,
+  ) {}
+
+  async insertProduct(title: string, desc: string, price: number) {
+    const newPeoduct = new this.productModel({
+      title,
+      description: desc,
+      price,
+    });
+    const res = await newPeoduct.save();
+
+    if (res) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  async getProduct() {
+    const result = await this.productModel.find().exec();
+    return result.map((product) => ({
+      id: product.id,
```

Comment:
Why is a map used here?
Write in comment.

Reply:
Because we want to get a simple array containing an object (Object have id in the string, not in new Object("") form). That's why I used a map function here.